src/main/java/de/tum/in/www1/artemis/service/plagiarism/ProgrammingPlagiarismDetectionService.java

```diff
@@ -330,14 +330,17 @@ public void deleteTempLocalRepository(Repository repository) {
         }
     }
 
-    private LanguageOption getJPlagProgrammingLanguage(ProgrammingExercise programmingExercise) {
-        return switch (programmingExercise.getProgrammingLanguage()) {
-            case JAVA -> LanguageOption.JAVA;
-            case C -> LanguageOption.C_CPP;
-            case PYTHON -> LanguageOption.PYTHON_3;
+    private Language getJPlagProgrammingLanguage(ProgrammingExercise programmingExercise) {
+        var languageIdentifier = switch (programmingExercise.getProgrammingLanguage()) {
```

Comment:
I would suggest to instantiate the Languages directly. The dynamic loading mechanism is not needed as far as I can see.